File: src/main/java/videostore/clean/Customer.java
package videostore.clean;
import java.util.ArrayList;
import java.util.List;

class Customer {
	private final String name;
	private List<Rental> rentals = new ArrayList<>();

	public Customer(String name) {
		this.name = name;
	}

	public void addRental(Rental rental) {
		rentals.add(rental);
	}

	public String getName() {
		return name;
	}

	public String getStatement() {
		double totalPrice = 0;
		int totalPoints = 0;
		String statement = getStatementHeader();
		for (Rental rental : rentals) {
			totalPoints += rental.getFrequentRenterPoints();
			totalPrice += rental.getPrice();
			statement += rental.getStatementLine();
		}
		statement += getStatementFooter(totalPrice, totalPoints);
		return statement;
	}

	private String getStatementHeader() {
		return "Rental Record for " + name + "\n";
	}


	private String getStatementFooter(double totalPrice, int totalFrequentRenterPoints) {
		return "Amount owed is " + totalPrice + "\n" 
				+ "You earned " + totalFrequentRenterPoints
				+ " frequent renter points";
	}

}

File: cs/1_Original/Soat.CleanCode.VideoStore.Original/Customer.cs
using System.Collections.Generic;
using System.Globalization;

namespace Soat.CleanCode.VideoStore.Original
{
    public class Customer
    {
        private readonly List<Rental> _rentals = new List<Rental>();
        public string Name { get; }

        public Customer(string name)
        {
            Name = name;
        }

        public void AddRental(Rental rental)
        {
            _rentals.Add(rental);
        }

        public string Statement()
        {
            var frequentRenterPoints = 0;
            var totalAmount = 0m;
            var result = "Rental Record for " + Name + "\r\n";
            foreach (var rental in _rentals)
            {
                decimal thisAmount = rental.GetAmmount();

                frequentRenterPoints++;

                if (rental.Movie.PriceCode == Movie.NEW_RELEASE
                    && rental.DaysRented > 1)
                {
                    frequentRenterPoints++;
                }

                result += "\t" + rental.Movie.Title + "\t" + thisAmount.ToString("0.0", CultureInfo.InvariantCulture) + "\r\n";
                totalAmount += thisAmount;
            }

            result += "You owed " + totalAmount.ToString("0.0", CultureInfo.InvariantCulture) + "\r\n";
            result += "You earned " + frequentRenterPoints.ToString() + " frequent renter points \r\n";

            return result;
        }

        
    }
}
